cachi2/core/package_managers/gomod.py

```diff
@@ -811,7 +816,49 @@ def _resolve_gomod(
         # Make Go ignore the vendor dir even if there is one
         go_list.extend(["-mod", "readonly"])
 
-    main_module_name = go([*go_list, "-m"], run_params).rstrip()
+    go_list_modules = go([*go_list, "-m", "-json"], run_params).rstrip()
+    module_list = list(load_json_stream(go_list_modules))
+    module_list_paths = [RootedPath(module["Dir"]) for module in module_list]
+
+    for index, module_path in enumerate(module_list_paths):
+        if app_dir.path == module_path.path:
+            raw_main_module = module_list.pop(index)
+            break
+    try:
+        main_module_name = raw_main_module["Path"]
+    except NameError:
+        raise PackageManagerError(f"Main module not found, it should have been at {app_dir.path}.")
+
+    modules_in_go_sum = _parse_go_sum_files(app_dir, module_list_paths)
+
+    def get_relative_path_workspace_to_main_module(path1: Path, path2: Path) -> str:
+        """Return relative path from workspace module to main module."""
+        if not path2.is_relative_to(app_dir.root):
+            raise PackageManagerError(
+                f"Workspace module path {path2} is not relative to root path {app_dir.root}."
+            )
+        relative_path = os.path.relpath(path2, path1)
+        return relative_path
```

Comment:
I believe we need to return `f"./{relative_path}"`. The reason for that is because the function that will convert this to a `Module` later on uses the dot to [identify](https://github.com/containerbuildsystem/cachi2/blob/d654bf3b763d56783d9bee6d65eaea9fbae58b07/cachi2/core/package_managers/gomod.py#L447) it's a local path.

You can try to reproduce the error by trying to prefetch this:

```
repo: https://github.com/kubernetes/ingress-nginx
ref: 3c4e78e6b755eb33821c4477106f424bd8792d8f
path: .
```

And then check the purl on the `k8s.io/ingress-nginx/magefiles` module. You'll see the version is `@magefiles`.